ddtrace/contrib/django/patch.py

```diff
@@ -106,17 +106,19 @@ def _set_request_tags(django, span, request):
 
     user = getattr(request, "user", None)
     if user is not None:
-        if hasattr(user, "is_authenticated"):
-            span.set_tag("django.user.is_authenticated", user_is_authenticated(user))
-
-        uid = getattr(user, "pk", None)
-        if uid:
-            span.set_tag("django.user.id", uid)
-
-        if config.django.include_user_name:
-            username = getattr(user, "username", None)
-            if username:
-                span.set_tag("django.user.name", username)
+        # Note: getattr calls to user / user_is_authenticated may result in ImproperlyConfigured exceptions from Django
```

Comment:
why?

Reply:
From the django framework test suite, it was found that in [test_closes_connection_with_content_length](https://sourcegraph.com/github.com/django/django/-/blob/tests/servers/tests.py#L217:9) the tracer library's django patch checks the `user` lazy object for its attributes and authentication. This authentication call actually triggers django's internal `get_user()` to grab the session key from the user object which doesn't exist, but this shouldn't happen (and doesn't get called without the tracer enabled) and results in a `ImproperlyConfigured` error that breaks the server.

Reply:
It might be worth adding a link to the Django core where this happens, just so when we come back in 6 months it'll be easy to remember why this is here.